bcipy/task/paradigm/matrix/calibration.py

```diff
@@ -208,35 +229,43 @@ def write_trigger_data(self, timing: List[Tuple[str, float]], first_run) -> None
             We do not write the calibration trigger used to generate this offset from the display.
             See MatrixDisplay._trigger_pulse() for more information.
         """
-        # write offsets. currently, we only check for offsets at the beginning.
-        if self.daq.is_calibrated and first_run:
-            self.trigger_handler.add_triggers(
-                [Trigger(
-                    'starting_offset',
-                    TriggerType.OFFSET,
-                    # offset will factor in true offset and time relative from beginning
-                    (self.daq.offset(self.matrix.first_stim_time) - self.matrix.first_stim_time)
-                )]
-            )
+        if first_run:
+            triggers = []
+            for content_type, client in self.daq.clients_by_type.items():
+                label = offset_label(content_type.name)
+                time = client.offset(
+                    self.matrix.first_stim_time) - self.matrix.first_stim_time
+                triggers.append(Trigger(label, TriggerType.OFFSET, time))
+            self.trigger_handler.add_triggers(triggers)
 
         # make sure triggers are written for the inquiry
         self.trigger_handler.add_triggers(convert_timing_triggers(timing, timing[0][0], self.trigger_type))
 
     def write_offset_trigger(self) -> None:

```

Comment:
Updated this to match the recent update to support multiple device offsets.. I think this is correct @lawhead

Reply:
Thanks, I just noticed this missing. Looks right to me.